nats.go

```diff
@@ -2610,14 +2672,26 @@ func (nc *Conn) PublishMsg(m *Msg) error {
 	if m == nil {
 		return ErrInvalidMsg
 	}
-	return nc.publish(m.Subject, m.Reply, m.Data)
+	var hdr []byte
+	if len(m.Header) > 0 {
+		if !nc.info.Headers {
```

Comment:
Seeing this, what happens to messages that are buffered during a disconnect and flushed on a reconnect, possibly to a server that does not support them. Will have to document that every server in a cluster is expected to be at same level.

Reply:
If we queued messages that had headers and we reconnect to a server that does not we will try to send them and get disconnected I believe.